### PR TITLE
Fix test flake TestNetworkUpdateTriggers

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Made a mistake in the `expectGateways`. /cc @hzxuzhonghu 

Fix https://github.com/istio/istio/pull/48157#issuecomment-1840042807

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/network_test.go [hunk 1]
```diff
@@ -77,9 +77,9 @@ func TestNetworkUpdateTriggers(t *testing.T) {
 		setGws(c.NetworkGateways())
 	})
 	expectGateways := func(t *testing.T, expectedGws int) {
+		// wait for a notification
+		assert.ChannelHasItem(t, notifyCh)
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/network_test.go:81]: Is notifyCh a must here? I do think we reply on `retry.UntilSuccessOrFail` to check the gw cb correctly executed.

**Review by howardjohn — APPROVED** [review]

**Review by jwendell — APPROVED** [review]
thanks. It was failing *all* the time for me locally. This fixes it for me.

File: pilot/pkg/serviceregistry/kube/controller/network_test.go (hunk 1 above)
Comment (hanxiaop) [reply pilot/pkg/serviceregistry/kube/controller/network_test.go:81]: It depends on the number of retries for the tests. If we have the channel, almost no retries will be performed. An interesting observation is that running 1000 tests with the notifyCh channel takes ~15 seconds, whereas it takes ~90 seconds without the channel.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/network_test.go:81]: That's weird, IMO the retry at most increase 10ms.

Both channel and retry need to wait for the `CB`.  If the cb will only be run once, the channel is the most efficient, if the cb should be run more times, then channel behaves same as the retry.

Comment (hanxiaop) [reply pilot/pkg/serviceregistry/kube/controller/network_test.go:81]: I just checked with a new environment, which has around 6 retries without the channel, and it resulted in an increase of 60ms. This is consistent with your statement. I guess my Mac environment is causing the unstable number of retries.

**Pull request merged**